File: CSharpScheduler/Employee.cs
using System.Collections.Generic;

namespace ShiftScheduler
{
    public class Employee
    {
        public string Name { get; set; }
        public Dictionary<string, string> PreferredShifts { get; set; } = new();
        public Dictionary<string, string> Schedule { get; set; } = new();
        public int AssignedDays => Schedule.Count;

        public Employee(string name)
        {
            Name = name;
        }

        public bool CanWork(string day)
        {
            return !Schedule.ContainsKey(day) && AssignedDays < 5;
        }

        public void AssignShift(string day, string shift)
        {
            Schedule[day] = shift;
        }
    }
}

File: PythonScheduler/employee.py
class Employee:
    def __init__(self, name):
        self.name = name
        self.preferred_shifts = {}  # e.g., {"Monday": "Morning", "Tuesday": "Evening", ...}
        self.assigned_days = 0
        self.schedule = {}  # final assigned schedule: {day: shift}

    def can_work(self, day):
        return day not in self.schedule and self.assigned_days < 5

    def assign_shift(self, day, shift):
        self.schedule[day] = shift
        self.assigned_days += 1
